handler/rfc8628/token_handler_test.go

```diff
@@ -0,0 +1,685 @@
+// Copyright © 2024 Ory Corp
+// SPDX-License-Identifier: Apache-2.0
+
+package rfc8628
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"testing"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/golang/mock/gomock"
+	"github.com/ory/fosite/internal"
+
+	"github.com/patrickmn/go-cache"
+
+	"github.com/ory/fosite/handler/oauth2"
+	"github.com/ory/fosite/token/hmac"
+
+	"github.com/ory/fosite"
+	"github.com/ory/fosite/storage"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+var hmacshaStrategy = oauth2.HMACSHAStrategy{
+	Enigma: &hmac.HMACStrategy{Config: &fosite.Config{GlobalSecret: []byte("foobarfoobarfoobarfoobarfoobarfoobarfoobarfoobar")}},
+	Config: &fosite.Config{
+		AccessTokenLifespan:   time.Hour * 24,
+		AuthorizeCodeLifespan: time.Hour * 24,
+	},
+}
+
+var RFC8628HMACSHAStrategy = DefaultDeviceStrategy{
+	Enigma: &hmac.HMACStrategy{Config: &fosite.Config{GlobalSecret: []byte("foobarfoobarfoobarfoobarfoobarfoobarfoobarfoobar")}},
+	RateLimiterCache: cache.New(
+		time.Hour*12,
+		time.Hour*24,
+	),
+	Config: &fosite.Config{
+		DeviceAndUserCodeLifespan: time.Hour * 24,
+	},
+}
+
+func TestDeviceUserCode_PopulateTokenEndpointResponse(t *testing.T) {
```

Comment:
TBH, I personally think the tests in this file are somehow duplicated with the ones in the `flow_authorize_code_auth_test.go`. I think all test cases should be merged into a `flow_generic_code_token_test.go`.

The `flow_authorize_code_auth_test` and `token_handler_test` should test their implementations of the interface `CodeTokenEndpointHandler`.

Let me know what you suggest.

Reply:
it is somehow normal practice (afaik) to have each file with a correspondent `_test.go` where u simply test in isolation that chunk of code

u might be able to try with a`go test -v ./ -cover -run <regexp>` where u set`regexp` to the tests u want to execute, then verify the coverage and see if they overlap or not

if they do i m happy to drop them in one of those places or rewrite the tests in a different way